provider/ec2/provider.go

```diff
@@ -189,38 +189,48 @@ your details are being verified.`
 // verify the configured credentials. If verification fails, a user-friendly
 // error will be returned, and the original error will be logged at debug
 // level.
-var verifyCredentials = func(e *environ) error {
+var verifyCredentials = func(e *environ, ctx context.ProviderCallContext) error {
 	_, err := e.ec2.AccountAttributes()
-	return maybeConvertCredentialError(err)
+	return maybeConvertCredentialError(err, ctx)
 }
 
 // maybeConvertCredentialError examines the error received from the provider.
 // Authentication related errors are wrapped in common.CredentialNotValid.
 // Authorisation related errors are annotated with an additional
 // user-friendly explanation.
 // All other errors are returned un-wrapped and not annotated.
-var maybeConvertCredentialError = func(err error) error {
+var maybeConvertCredentialError = func(err error, ctx context.ProviderCallContext) error {
 	if err == nil {
 		return nil
 	}
 
+	convert := func(converted error) error {
+		logger.Errorf("XXXXXXXXXXXXXXXXXXX calling invalidate credential")
```

Comment:
Debugging output?

Reply:
Oh jeez... what gave it away? :D 
Yes, thank you - it is not needed in prod code.